integration-tests/src/metadata/validation.rs

```diff
@@ -76,9 +76,11 @@ async fn full_metadata_check() {
     assert!(api.validate_metadata().is_ok());
 
     // Modify the metadata.
-    let locked_client_metadata = api.client.metadata();
-    let client_metadata = locked_client_metadata.read();
-    let mut metadata: RuntimeMetadataV14 = client_metadata.runtime_metadata().clone();
+    let mut metadata: RuntimeMetadataV14 = {
+        let locked_client_metadata = api.client.metadata();
```

Comment:
Could this just be `let client_metadata = api.client.metadata().read();`

Reply:
It was, but the problem with that is that you acquire a lock for the rest of the block, which means it's held across `.await`'s (which is bad; can elaborate if you want) and clippy complains about it :)

Reply:
Okay yea that totally makes sense. I assumed their might be some asynchronous process happening! Thanks for the clarification.